lib/queryBuilder/QueryBuilder.js

```diff
@@ -325,7 +325,7 @@ class QueryBuilder extends QueryBuilderBase {
         );
       } else if (op.constructor === FromOperation) {
         return op.table === tableName;
-      } else if (op.name === 'as' || op.is(FindOperation)) {
+      } else if (op.name === 'as' || op.is(FindOperation) || op.is(OnErrorOperation)) {
```

Comment:
I have no recollection though why I added this.

Reply:
It's not wrong: If you have a query that only contains an `onError()` operation, it's still a select-all query.
As for why it was added, I'd guess you had some logic in your code base that always calls `onError()` o your queries and wanted to filter this out? So I do think this makes sense, but it's not really in the scope of this PR to add it?